src/Exceptions/ExceptionWithContext.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Cascade\Exceptions;
+
+/**
+ * A custom exception class which provides a context argument for later reference.
+ */
+class ExceptionWithContext extends \Exception
```

Comment:
I think my feeling was about the name actually. `ExceptionWithContext` doesn't describe what the exception is, or what happened. When you catch an exception, doesn't it help if that exception pertains to a particular problem, like `ValidationException` (must have happened during validation), `RangeException` (got too many of something), but `ExceptionWithContext` doesn't tell you anything. I guess neither does `Exception`, but that's exactly why we don't throw plain old `Exception` very often.

Keep it as is, it's good nuff. I imagine you'll mostly be using it as a base to extend from though.